### ISSUE TITLE
Ocad export/ import - real word coordinates angle/ map north

### ISSUE THREAD

**Issue by ollesmaps** [OpenOrienteering/mapper#1896]
### Steps to reproduce
1. Import into OOM ocd (real word coordinates angle has many digits 4,0821 / it is possible to have in Ocad up to 8 digits after comma).
2. Export the same file from OOM into ocd
3. The real word coordinates angle in Ocad is now 4,08 (maximum 2 digits after comma).

### Expected behaviour
The angle shall be the same

### Configuration
Mapper Version: 0.9.3
Operating System: Win, Android

**Comment by lpechacek** [comment]
There is no rounding applied to the number in the OCD import and export path. Would you mind sharing a demo file for debugging, please?

By the way, did you use the map georeferencing dialog with this map? I can imagine that it might be the source of the rounding.

**Comment by ollesmaps** [comment]
I did not modify any numbers. Just plain import/ export.
Sample attached, numbered in order of exports:
[kost_deklinace_oc12_2.zip](https://github.com/OpenOrienteering/mapper/files/6130215/kost_deklinace_oc12_2.zip)

**Comment by dg0yt** [comment]
Class `Georeferencing` applies `roundDeclination()` to the arguments given to the setters for declination/grivation, and OCD import must go through these setters.

https://github.com/OpenOrienteering/mapper/blob/9e874fe7d9d1b19e7560a2ed63be9a5da5cd0908/src/core/georeferencing.cpp#L841-L853

**Comment by pkturner** [comment]
While I agree that it would sometimes be helpful if Mapper preserved all the digits provided by OCD, there are good reasons for rounding to two decimal places.

- Avoid complexity in the Georeferencing dialog.
- Give the user control over all the decimal digits that Mapper respects.

Perhaps Mapper could change its rounding of grivation to be 8 places, while still rounding declination to 2.

**Comment by ollesmaps** [comment]
Is this rather?
- rounding
- or automatic rotation to that new degree?

In Ocad it triggers this:
![angle](https://user-images.githubusercontent.com/20535899/111062582-0e0d6e00-84aa-11eb-8b08-967f703d6ab6.jpg)

**Comment by lpechacek** [comment]
> Is this rather?
> 
>     * rounding

Yes, it's pure rounding of the numeric value. There is no rotation applied to match the rounded value. The OCAD warning would be a reason to rectify the export issue. We should strive for exporting the same information bits that we receive where possible.

@pkturner, @dg0yt, I'm not a big fan of increasing the number of decimal places in the georeferencing dialog. Mapper is not a geodetic software, unlike OCAD, and the best tools the mappers use can measure azimuths with 0.1° precision. So, the two decimal places are just right, IMHO.

Ad the import/export matter, can you please increase the number of digits preserved in `setGrivation()` to eight?

For the record, this is how the `a` parameter changes in string 1039 on export.
* loaded from .ocd: _m10000.000000	g25.0000	r1	x507000	y5594000	**a4.08210000**	d250.00	b0.00	c0.00	i2033_
* saved to .ocd: _m10000.000000	g25.0000	r1	x507000	y5594000	**a4.08000000**	d250.00	i2033	b0.00	c0.00_

**Comment by ollesmaps** [comment]
Perhaps a _notice_ rounding was applied is enough. 

(The above mentioned Ocad dialogue box appears in last exported file when you try to open in Ocad original ocd as background map.)
